From 8a5e5df92dacd694208a4ac22ca16b7470acebfb Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 28 Mar 2017 14:47:46 +0800
Subject: [PATCH 1649/5242] MLK-14286-3 ARM: dts: imx7ulp-evk-hdmi: create the
 hdmi dts file

commit  6012ab3af4c1f786e71f5ce288e4a4e7c8c37fe7 from
https://source.codeaurora.org/external/imx/linux-imx.git

Create a new dts file 'imx7ulp-evk-hdmi.dts' to enable hdmi
display to avoids conflict with mipi dsi panel display. Use
endpoint to connect dsi controller and adv7535.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 86fb9340d01a4583628d465f29fba67df93c453c)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile             |    3 ++-
 arch/arm/boot/dts/imx7ulp-evk-hdmi.dts |   26 ++++++++++++++++++++++++++
 2 files changed, 28 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/boot/dts/imx7ulp-evk-hdmi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 402d656..3bfeff1 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -706,7 +706,8 @@ dtb-$(CONFIG_SOC_IMX7ULP) += \
 	imx7ulp-evk-sd1.dtb \
 	imx7ulp-evk-lpuart.dtb \
 	imx7ulp-evk-qspi.dtb \
-	imx7ulp-evk-wm8960.dtb
+	imx7ulp-evk-wm8960.dtb \
+	imx7ulp-evk-hdmi.dtb
 dtb-$(CONFIG_SOC_LS1021A) += \
 	ls1021a-moxa-uc-8410a.dtb \
 	ls1021a-qds.dtb \
diff --git a/arch/arm/boot/dts/imx7ulp-evk-hdmi.dts b/arch/arm/boot/dts/imx7ulp-evk-hdmi.dts
new file mode 100644
index 0000000..d56c8e1
--- /dev/null
+++ b/arch/arm/boot/dts/imx7ulp-evk-hdmi.dts
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2017 NXP.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx7ulp-evk.dts"
+
+&adv7535 {
+	status = "okay";
+	port {
+		dsi_to_hdmi: endpoint {
+			remote-endpoint = <&mipi_dsi_ep>;
+		};
+	};
+};
+
+&mipi_dsi {
+	port {
+		mipi_dsi_ep: endpoint {
+			remote-endpoint = <&dsi_to_hdmi>;
+		};
+	};
+};
-- 
1.7.9.5

